Small size job executed twice occasionally

## Description

Sometimes a small size job (i.e. a job with very short execution time) is executed twice in sequential ticks.

## Reproducing Steps

1. Go to https://github.com/jasonlvhit/gocron/blob/98b59b546dee5e226421a00e3cfcb6223bdc839e/gocron_test.go#L36
2. Increase `numberOfIterations` like 20 (10x times)
3. Run `TestSeconds()`
4. Below is my failure logs:
<details><summary>Failure Logs</summary>

```
GOROOT=/usr/local/Cellar/go/1.13.1/libexec #gosetup
GOPATH=/Users/hangpark/Projects/go #gosetup
/usr/local/Cellar/go/1.13.1/libexec/bin/go test -c -o /private/var/folders/0s/mw5ww6q52t5fmc1s9nsn_01w0000gp/T/___TestSeconds_in_github_com_jasonlvhit_gocron github.com/jasonlvhit/gocron #gosetup
/usr/local/Cellar/go/1.13.1/libexec/bin/go tool test2json -t /private/var/folders/0s/mw5ww6q52t5fmc1s9nsn_01w0000gp/T/___TestSeconds_in_github_com_jasonlvhit_gocron -test.v -test.run ^TestSeconds$ #gosetup
=== RUN   TestSeconds
--- FAIL: TestSeconds (32.00s)
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations


Expected :2
Actual   :1
<Click to see difference>


FAIL

Process finished with exit code 1
```

</details>

## Why?

For each tick, the scheduler calculates jobs to run at that tick by calling `getRunnableJobs()`. This  method is just call `job.shouldRun()` for all registered jobs, which is:
https://github.com/jasonlvhit/gocron/blob/98b59b546dee5e226421a00e3cfcb6223bdc839e/gocron.go#L102-L104

And then the scheduler runs runnable jobs (`job.run()`). After each runnable job is executed, that is at the end of `job.run()`, a job is rescheduled by
https://github.com/jasonlvhit/gocron/blob/98b59b546dee5e226421a00e3cfcb6223bdc839e/gocron.go#L128

But, in `job.run()`, **if current time is BEFORE next running time, it skips rescheduling.**
https://github.com/jasonlvhit/gocron/blob/98b59b546dee5e226421a00e3cfcb6223bdc839e/gocron.go#L291-L293

Now, let me illustrate a sample case:

1. Job named `job` has `nextRun` at `2020-01-01T00:00:00.900Z`
2. Tick is started at `2020-01-01T00:00:00.000Z`
3. `job` is appended into runnable jobs **<----- (A)**
4. Scheduler executes `job`
5. `job` has been executed very fast, current time: `2020-01-01T00:00:00.100Z`
6. `job` should be rescheduled
7. But `job.nextRun` is after the current time. **<------ (B)**
8. On next second, the next tick append `job` into runnable jobs
9. `job` is executed TWICE!

The problem is, **(A)** compares `nextRun` with the current time by using "seconds" while **(B)** compares by using "nanoseconds". In other words, `time.Now().Unix() >= j.nextRun.Unix()` and `j.nextRun.After(now)` have different degrees of precision.

## Does this fix?

Is this bug fixed if we change `time.Now().Unix() >= j.nextRun.Unix()` to `time.Now.After(j.nextRun)`? Yeah, probabily... but with this fix may cause the another bug.

Since every function has at least nanosecond execution time, `job := sched.Every(2).Seconds()` is sometimes executed in 3 secs, not 2. See below test results:

<details><summary>After applying ns precision</summary>

```
=== RUN   TestSeconds
--- FAIL: TestSeconds (48.00s)
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations
    gocron_test.go:53: 
        	Error Trace:	gocron_test.go:53
        	            				gocron_test.go:30
        	Error:      	Not equal: 
        	            	expected: 2
        	            	actual  : 3
        	Test:       	TestSeconds
        	Messages:   	Duration between tasks does not correspond to expectations


Expected :2
Actual   :3
<Click to see difference>


FAIL

Process finished with exit code 1
```

</details>

## TODO

Can changing tick interval more finely be the solution?
Or can we just delete below code? https://github.com/jasonlvhit/gocron/blob/98b59b546dee5e226421a00e3cfcb6223bdc839e/gocron.go#L291-L293

Anyway, this bug seems to be critical for cases which are non-idempotent jobs (with fast execution time).

--- a/gocron_test.go
+++ b/gocron_test.go
@@ -57,10 +57,17 @@ func testJobWithInterval(t *testing.T, sched *Scheduler, job *Job, expectedTimeB
 func TestSafeExecution(t *testing.T) {
 	sched := NewScheduler()
 	success := false
+	var wc sync.WaitGroup
+	wc.Add(1)
+
 	sched.Every(1).Second().Do(func(mutableValue *bool) {
 		*mutableValue = !*mutableValue
+		wc.Done()
 	}, &success)
+
 	sched.RunAll()
+	wc.Wait()
+
 	assert.Equal(t, true, success, "Task did not get called")
 }
 

--- a/job.go
+++ b/job.go
@@ -63,11 +63,6 @@ func (j *Job) run() ([]reflect.Value, error) {
 		locker.Lock(key)
 		defer locker.Unlock(key)
 	}
-
-	j.lastRun = time.Now()
-	if err := j.scheduleNextRun(); err != nil {
-		return nil, err
-	}
 	result, err := callJobFuncWithParams(j.funcs[j.jobFunc], j.fparams[j.jobFunc])
 	if err != nil {
 		return nil, err
@@ -94,7 +89,11 @@ func (j *Job) Do(jobFun interface{}, params ...interface{}) error {
 	j.funcs[fname] = jobFun
 	j.fparams[fname] = params
 	j.jobFunc = fname
-	j.scheduleNextRun()
+
+	now := time.Now().In(j.loc)
+	if !j.nextRun.After(now) {
+		j.scheduleNextRun()
+	}
 
 	return nil
 }
@@ -199,10 +198,6 @@ func (j *Job) scheduleNextRun() error {
 	now := time.Now()
 	if j.lastRun == time.Unix(0, 0) {
 		j.lastRun = now
-	}
-
-	if j.nextRun.After(now) {
-		return nil
 	}
 
 	periodDuration, err := j.periodDuration()

--- a/scheduler.go
+++ b/scheduler.go
@@ -87,7 +87,9 @@ func (s *Scheduler) RunPending() {
 
 	if n != 0 {
 		for i := 0; i < n; i++ {
-			runnableJobs[i].run()
+			go runnableJobs[i].run()
+			runnableJobs[i].lastRun = time.Now()
+			runnableJobs[i].scheduleNextRun()
 		}
 	}
 }
@@ -100,7 +102,7 @@ func (s *Scheduler) RunAll() {
 // RunAllwithDelay runs all jobs with delay seconds
 func (s *Scheduler) RunAllwithDelay(d int) {
 	for i := 0; i < s.size; i++ {
-		s.jobs[i].run()
+		go s.jobs[i].run()
 		if 0 != d {
 			time.Sleep(time.Duration(d))
 		}

